### PR TITLE
chore(atlas-service): add validation for backend response

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch makes the backend response validation stricter, adds some new tests and adjusts existing ones to account for the change

### CODE REVIEW

**Review by Anemy — APPROVED** [review]

File: packages/atlas-service/src/util.ts [hunk 1]
```diff
@@ -11,16 +11,113 @@ export type IntrospectInfo = { active: boolean };
...
+        )}`
+      );
+    }
+  }
+
+  if (query.aggregation && typeof query.aggregation.pipeline !== 'string') {
+    throw new Error(
+      `Unexpected response: expected aggregation pipeline to be a string, got ${String(
+        query.aggregation
+      )}`
```
Comment (addaleax) [inline packages/atlas-service/src/util.ts:118]: Could we use `util.inspect()` here? It sounds like we’re expecting `query.aggregation` to be an object, so what we’d be getting by default if it just happens to be an object of a different shape is `[object Object]` here (ditto above)

File: packages/atlas-service/src/util.ts [hunk 2]
```diff
@@ -11,16 +11,113 @@ export type IntrospectInfo = { active: boolean };
 
 export type Token = plugin.IdPServerResponse;
 
+function hasExtraneousKeys(obj: any, expectedKeys: string[]) {
+  return (
+    Object.keys(obj).filter((key) => !expectedKeys.includes(key)).length > 0
```
Comment (addaleax) [inline packages/atlas-service/src/util.ts:16]: tiny and very optional suggestion:

```suggestion
    Object.keys(obj).some((key) => !expectedKeys.includes(key))
```

File: packages/atlas-service/src/main.spec.ts [hunk 3]
```diff
@@ -243,10 +260,28 @@ describe('AtlasServiceMain', function () {
...
           '{"userInput":"test","collectionName":"jam","databaseName":"peanut","schema":{"_id":{"types":[{"bsonType":"ObjectId"}]}},"sampleDocuments":[{"_id":1234}]}'
         );
-        expect(res).to.have.nested.property(
-          'content.query.filter',
-          "{ test: 'pineapple' }"
-        );
+        expect(res).to.deep.eq(responses.success);
+      });
+
+      it('should fail when response is not matching expectec schema', async function () {
```
Comment (addaleax) [inline packages/atlas-service/src/main.spec.ts:266]:
```suggestion
      it('should fail when response is not matching expected schema', async function () {
```

**Review by addaleax — APPROVED** [review]

File: packages/atlas-service/src/util.ts (hunk 1 above)
Comment (gribnoysup) [reply packages/atlas-service/src/util.ts:118]: Yeah, good idea, haven't thought about that, will update

**Pull request merged**